xml/operating_system.xml

```diff
@@ -680,6 +683,7 @@
     <description>HP OSes</description>
     <example os.product="HP-UX" os.family="HP-UX">HP-UX</example>
     <example os.product="OpenVMS" os.family="OpenVMS">OpenVMS</example>
+    <example os.product="OpenVMS" os.family="OpenVMS" os.version="1.0">OpenVMS 1.0</example>
```

Comment:
Is this a valid version number? All of the versions listed on [OpenVMS Releases](https://wiki.vmssoftware.com/OpenVMS#Releases) start with a "V".

Reply:
I noticed the comment on the PR description that stated these were "made up to satisfy the regular expression".

Reply:
We should ensure that all examples have provenance.

Reply:
Correct, I made this up -- though 1.0 is a valid version. Generally versions should be in numeric dotted notation for easier comparison ("is 1.2.3 less than 4.0"? is programmatically easier to ask than "is V1.2.3 less than 4.0?" -- which is done often for vulnerability testing), though I do see other places in recog that extract stuff like "V" before the version number. The exception to that rule is when you have old Cisco IOS versions like 12(T)13b (or whatever, I've really tried to erase those from my memory) where they don't really follow a dotted notation format. But if the purpose of "V" is only to denote that it is a version, there's no point in keeping it as part of the extracted version string IMHO.

Reply:
Correct, I made this up -- though 1.0 is a valid version. Generally versions should be in numeric dotted notation for easier comparison ("is 1.2.3 less than 4.0"? is programmatically easier to ask than "is V1.2.3 less than 4.0?" -- which is done often for vulnerability testing), though I do see other places in recog that extract stuff like "V" before the version number. The exception to that rule is when you have old Cisco IOS versions like 12(T)13b (or whatever, I've really tried to erase those from my memory) where they don't really follow a dotted notation format. But if the purpose of "V" is only to denote that it is a version, there's no point in keeping it as part of the extracted version string IMHO.

Reply:
Having issues today with github single comments & reviews 🤦

Reply:
Also note that "V1.0" wouldn't match the regular expression without modifying the regex to match a V. Just because it's listed in Wikipedia that way doesn't mean that was the originally intended way for it to be represented -- in fact I'd say it wasn't, since the original author (old schooler like myself or tas at rapid7) didn't include it

Reply:
I agree the dotted notation portion makes sense for the parameter value. It does appear that the fingerprint regex is not correct. The table of version numbers that I linked was from a VMS Software wiki not from Wikipedia. I found documentation on [confirming the installation of the OpenVMS operating system](https://www.ibm.com/docs/en/ds8800?topic=attachment-confirming-installation-openvms-operating-system) that shows the version number reported with a leading "V".

```
$ show system
OpenVMS V7.3-2 on node DS20 5-FEB-2004 15:15:16.88 Uptime 1 01:45:09
```

Reply:
There is also documentation here:
https://marc.vos.net/books/vms/help/show/

That shows version 6.2 returning as:
```
$ SHOW SYSTEM
     OpenVMS 6.2 on node KRYPTN 14-DEC-1994 17:45:47:78 Uptime 2 21:53:59
```

However, regardless, this patch wasn't meant to make behavioral changes to any of the regular expressions. I don't think this regular expression was designed to process output from the `show system` command directly, as it is very limited in what it would match. Thus, I say we keep it as is, but happy to change 1.0 to 6.2.

Reply:
I'll add that I'd be fine adding a _conditional_ "V?" to the regular expression (not captured within the version capture group), but again, this regex doesn't appear to have been designed to specifically parse the output of `show system` (as it anchors the end of the string after the version, whereas the real output of `show system` continues on with a timestamp etc.).

Reply:
Interesting find on the `SHOW SYSTEM` output. Thanks for making this one more flexible.